src/approx_equal/mod.rs

```diff
@@ -0,0 +1,38 @@
+#![allow(clippy::unused_unit)]
+use polars::prelude::*;
+use pyo3_polars::derive::polars_expr;
+use pyo3_polars::export::polars_core::export::num::Signed;
+use serde::Deserialize;
+
+#[derive(Deserialize)]
+pub struct ThresholdKwargs {
+    pub threshold: f64,
+}
+
+fn impl_approx_equal_numeric<T>(ca: &ChunkedArray<T>, threshold: T::Native) -> BooleanChunked
+where
+    T: PolarsNumericType,
+    T::Native: Signed,
+{
+    ca
+    .into_iter()
+    .map(|opt_v| opt_v.map(|v| v.abs() <= threshold))
+    .collect()
+}
+
+#[polars_expr(output_type=Boolean)]
+fn approx_equal(inputs: &[Series], kwargs: ThresholdKwargs) -> PolarsResult<Series> {
+    let s = &inputs[0] - &inputs[1];
+    let threshold = kwargs.threshold;
+    // need to handle the precision better... just cast to float when receive?
+    match s.dtype() {
+        DataType::Int32 => Ok(impl_approx_equal_numeric(s.i32().unwrap(), threshold.floor() as i32).into_series()),
+        DataType::Int64 => Ok(impl_approx_equal_numeric(s.i64().unwrap(), threshold.floor() as i64).into_series()),
+        DataType::Float32 => Ok(impl_approx_equal_numeric(s.f32().unwrap(), threshold as f32).into_series()),
+        DataType::Float64 => Ok(impl_approx_equal_numeric(s.f64().unwrap(), threshold).into_series()),
+        dtype => {
```

Comment:
Is there a reason why we aren't implementing this for all numeric types (except Decimal)?

Reply:
This is a fair point, I'll do it for the 16 and U types too.

Reply:
An update:
* I propose not supporting UInts for this operation. It will lead to unexpected behaviour with the diff process if it goes to zero. Although we could handle this, it's not very common to see UInts without explicitly setting the number to be UInt in the schema, and in that case maybe you are using the wrong dtype if you are doing operations like this.
* The Int16 panics when I tried to run it in tests (error message below). Marco Gorelli didn't include it in [his tutorial,](https://marcogorelli.github.io/polars-plugins-tutorial/abs/). Int16 isn't behind a polars feature that can be enabled, so I expect Int16 just isn't supported for polars plugins. Or it's a bug in polars?...

```
panicked at src/approx_equal/mod.rs:23:1:
called `Result::unwrap()` on an `Err` value: ComputeError(ErrString("cannot create series from Int16"))
```

Reply:
Good point for UInt -- it probably doesn't come up too much.
Not sure about the Int16 🤷 Maybe just leave it for now, and come back to it when we try out other stuff?